resources/assets/lib/beatmap-discussions/beatmap-list.tsx

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import Blackout from 'blackout';
+import BeatmapJsonExtended from 'interfaces/beatmap-json-extended';
+import * as React from 'react';
+import { classWithModifiers } from 'utils/css';
+import BeatmapListItem from './beatmap-list-item';
+
+interface Props {
+  beatmaps: BeatmapJsonExtended[];
+  createLink: (beatmap: BeatmapJsonExtended) => string;
+  currentBeatmap: BeatmapJsonExtended;
+  getCount?: (beatmap: BeatmapJsonExtended) => number | undefined;
+  onSelectBeatmap: (beatmapId: number) => void;
+}
+
+interface State {
+  showingSelector: boolean;
+}
+
+export default class BeatmapList extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      showingSelector: false,
+    };
+  }
+
+  componentDidMount() {
+    $(document).on('click.beatmapList', this.onDocumentClick);
```

Comment:
conflicts (the event listeners are missing unique id)

Reply:
Done